plugins/src/main/java/com/google/fhir/gateway/plugin/PatientAccessChecker.java

```diff
@@ -108,6 +109,16 @@ private AccessDecision processPatch(RequestDetailsReader requestDetails) {
     return checkNonPatientAccessInUpdate(requestDetails, HTTPVerb.PATCH);
   }
 
+  private AccessDecision processDelete(RequestDetailsReader requestDetails) {
+    // This AccessChecker does not allow deletion of Patient resource
+    if (FhirUtil.isSameResourceType(requestDetails.getResourceName(), ResourceType.Patient)) {
```

Comment:
We let updates to the `Patient` resource, why shouldn't we let deletion (I agree this is probably a rare edge case)?

Reply:
This access checker is intended for Patient apps. 
So If I am logged in as a patient then should I be allowed to delete my own patient ? Delete operation of a patient can cause invalid state in the application.

I can change this if you feel otherwise.

Reply:
Yeah, I see your point, but the actual patient-id can come from another source (e.g., Keycloak). So the scenario I was thinking about is that the patient want to delete their record from a system and for whatever reason the app and the FHIR server allow this to happen.
Again, this is a rare edge case, let's just go with the solution you have here to be safe for now.